go_sequencer/txprocessor/txprocessor.go

```diff
@@ -153,5 +177,1068 @@ func NewTxProcessor(state *statedb.StateDB, config Config) *TxProcessor {
+		}
+	}
+
+	// ZKInputs
+	if txProcessor.zki != nil {
+		// Txs
+		txProcessor.zki.TxCompressedData[txProcessor.txIndex], err = tx.TxCompressedData(txProcessor.config.ChainID)
+		if err != nil {
+			return nil, nil, false, common.Wrap(err)
+		}
+		txProcessor.zki.TxCompressedDataV2[txProcessor.txIndex], err = tx.TxCompressedDataV2()
+		if err != nil {
+			return nil, nil, false, common.Wrap(err)
+		}
+		txProcessor.zki.FromIdx[txProcessor.txIndex] = tx.FromIdx.BigInt()
+		txProcessor.zki.ToIdx[txProcessor.txIndex] = tx.ToIdx.BigInt()
+
+		// fill AuxToIdx if needed
+		if tx.ToIdx == 0 {
+			// use toIdx that can have been filled by tx.ToIdx or
+			// if tx.Idx==0 (this case), toIdx is filled by the Idx
+			// from db by ToEthAddr&ToBJJ
+			txProcessor.zki.AuxToIdx[txProcessor.txIndex] = tx.AuxToIdx.BigInt()
+		}
+
+		if tx.ToBJJ != common.EmptyBJJComp {
+			_, txProcessor.zki.ToBJJAy[txProcessor.txIndex] = babyjub.UnpackSignY(tx.ToBJJ)
+		}
+		txProcessor.zki.ToEthAddr[txProcessor.txIndex] = common.EthAddrToBigInt(tx.ToEthAddr)
+
+		txProcessor.zki.OnChain[txProcessor.txIndex] = big.NewInt(0)
+		amountF40, err := common.NewFloat40(tx.Amount)
+		if err != nil {
+			return nil, nil, false, common.Wrap(err)
+		}
+		txProcessor.zki.AmountF[txProcessor.txIndex] = big.NewInt(int64(amountF40))
+		txProcessor.zki.NewAccount[txProcessor.txIndex] = big.NewInt(0)
+		txProcessor.zki.MaxNumBatch[txProcessor.txIndex] = big.NewInt(int64(tx.MaxNumBatch))
+
+		// Rq fields: set zki to link the requested tx
+		if tx.RqOffset != 0 {
+			if tx.RqOffset > 7 { //nolint:gomnd
+				return nil, nil, false, common.New(ErrInvalidRqOffset)
+			}
+			rqOffset := big.NewInt(int64(tx.RqOffset))
+			txProcessor.zki.RqOffset[txProcessor.txIndex] = rqOffset
+			txProcessor.zki.RqTxCompressedDataV2[txProcessor.txIndex], err = tx.RqTxCompressedDataV2()
+			if err != nil {
+				return nil, nil, false, common.Wrap(err)
+			}
+			if tx.RqToBJJ != common.EmptyBJJComp {
+				_, txProcessor.zki.RqToBJJAy[txProcessor.txIndex] = babyjub.UnpackSignY(tx.RqToBJJ)
+			}
+			txProcessor.zki.RqToEthAddr[txProcessor.txIndex] = common.EthAddrToBigInt(tx.RqToEthAddr)
+		}
+
+		signature, err := tx.Signature.Decompress()
+		if err != nil {
+			log.Error(err)
+			return nil, nil, false, common.Wrap(err)
+		}
+		txProcessor.zki.S[txProcessor.txIndex] = signature.S
+		txProcessor.zki.R8x[txProcessor.txIndex] = signature.R8.X
+		txProcessor.zki.R8y[txProcessor.txIndex] = signature.R8.Y
+	}
+
+	// if StateDB type==TypeSynchronizer, will need to add Nonce
+	if txProcessor.state.Type() == statedb.TypeSynchronizer {
+		// as tType==TypeSynchronizer, always tx.ToIdx!=0
+		acc, err := txProcessor.state.GetAccount(tx.FromIdx)
+		if err != nil {
+			log.Errorw("GetAccount", "fromIdx", tx.FromIdx, "err", err)
+			return nil, nil, false, common.Wrap(err)
+		}
+		tx.Nonce = acc.Nonce
+	}
+
+	switch tx.Type {
+	case common.TxTypeTransfer, common.TxTypeTransferToEthAddr, common.TxTypeTransferToBJJ:
```

Comment:
For L2Tx will we have transfer ? Instead here will there we transactions related to vouch - Create, Delete

Let me know if i missed something ?

Reply:
No need of transfer, as you commented we just use vouch related txs.